### PR TITLE
Add mac osx compatibility with instructions

### PR DESCRIPTION
**Pull request by tylerlevine**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,18 @@ attachment of arbitrary data to an existing commit.
...
+
+Since the built in getopt and date binaries are not compatible with this script, you must install some GNU compatible binaries (homebrew recommended).
+
+  * gnu-getopt
+    * `brew install gnu-getopt`
+    * `ln -s $(brew --prefix gnu-getopt)/bin/getopt /usr/local/bin/gnu-getopt`
+  * coreutils
+    * `brew install coreutils`
+
+Make sure that the `gnu-getopt`, `gdate`, and `gshuf` binaries are in your $PATH.
```
Comment (lrvick) [inline README.md]: brew link should cover this too yeah?

File: bin/git-signatures [hunk 2]
```diff
@@ -214,7 +218,7 @@ path_check() {
 epoch_to_human() {
```
Comment (lrvick) [inline bin/git-signatures]: now that you are using gnu_date you can probably eliminate this entire function which only existed for osx compat.

File: README.md [hunk 3]
```diff
@@ -19,6 +19,18 @@ attachment of arbitrary data to an existing commit.
   * Git 2.1+
   * GnuPG 2.2+
 
+## Mac OS X ##
+
+Since the built in getopt and date binaries are not compatible with this script, you must install some GNU compatible binaries (homebrew recommended).
+
+  * gnu-getopt
+    * `brew install gnu-getopt`
+    * `ln -s $(brew --prefix gnu-getopt)/bin/getopt /usr/local/bin/gnu-getopt`
```
[the remark this answers is not in the document]
Comment (tylerlevine) [reply README.md]: It doesn't work (not sure why):

```
$ gnu-getopt
zsh: command not found: gnu-getopt
$ brew link --force gnu-getopt
Linking /usr/local/Cellar/gnu-getopt/1.1.6... 28 symlinks created

If you need to have this software first in your PATH instead consider running:
  echo 'export PATH="/usr/local/opt/gnu-getopt/bin:$PATH"' >> ~/.zshrc
$ gnu-getopt
zsh: command not found: gnu-getopt
$ ln -s $(brew --prefix gnu-getopt)/bin/getopt /usr/local/bin/gnu-getopt
$ gnu-getopt
getopt: missing optstring argument
Try `getopt --help' for more information.
```

[the remark this answers is not in the document]
Comment (lrvick) [reply README.md]: Whelp. Fair enough.

**Pull request merged**